Hametask 3

IgorShahin opened the pull request:
Добавлен FileService для работы с файлами и исключение FileOperationException для обёртки IO-ошибок

IgorShahin committed:
feat: add FileService and custom FileOperationException

IgorShahin committed:
update main

Remany commented on src/main/java/ru/aston/hometask/exception/FileOperationException.java:
```diff
@@ -0,0 +1,11 @@
+package ru.aston.hometask.exception;
+
+public class FileOperationException extends Exception {
```
для большой явности стоит наследовать от IOException

Remany commented on src/main/java/ru/aston/hometask/service/FileService.java:
```diff
@@ -0,0 +1,109 @@
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class FileService {
```
Дз можно было бы решить след образом: два класса FileReaderService и FileWriterService, реализуют интерфейс FileService, у которого всего один абстрактный метод process. Остальные ютильные методы, можно сделать дефолтными. И каждая из реализаций - производит свой процессинг файла (FileReaderService - читает, а FileWriterService - пишет)

Remany commented on src/main/java/ru/aston/hometask/Main.java:
```diff
@@ -0,0 +1,22 @@
+
+public class Main {
+    public static void main(String[] args) {
+        FileService fileService = new FileService("/root/protected.txt");
```
Вот таким образом путь к файлу лучше не передавать. Вместо этого лучше сделать небольшой интерактив и с помощью класса Scanner передавать путь к файлу. Но вообще обычно это делается через переменные окружения

Remany commented on src/main/java/ru/aston/hometask/Main.java:
```diff
@@ -0,0 +1,22 @@
+
+            if (e.getCause() != null) {
+                System.err.println("Original cause: " + e.getCause().getClass().getSimpleName());
+                System.err.println("Cause message: " + e.getCause().getMessage());
```
вместо вывода в консоль, начинай использовать логи

IgorShahin committed:
commit: refactor file service based on review comments
- FileOperationException now extends IOException
- introduced FileService interface with single abstract method process() and default utility methods
- added FileReaderService and FileWriterService implementations for read/write operations
- replaced System.out/err with java.util.logging

IgorShahin committed:
delete old FileService
